web/src/app/services/data-store/data-store.service.ts

```diff
@@ -486,6 +486,8 @@ export class DataStoreService {
    * Add a loiTask as first element, reindex the others.
    */
   addLoiTask(tasks: Map<string, Task>): Map<string, Task> {
+    if (tasks.find(task => !!task.addLoiTask)) return tasks;
```

Comment:
I think `some()` is more idiomatic in this case. Also, it may help readability a bit to make `null` check explicit:

`if (tasks.some(task => task.addLoiTask === true)) return tasks;`